syntax = "proto3";

option objc_class_prefix = "RAFT";

package raft;

// Request Vote message
message VoteRequest {
  string candidate_id = 1;
  uint64 candidate_term_id = 2;
  string dest_id = 3;
  uint64 last_log_index = 4;
  uint64 last_log_term = 5;
}

// Reply to the vote requst
message VoteReply {
  uint64 responder_term_id = 1;
  bool vote_granted = 2;
  string responder_uuid = 3;
}

message HeartBeat {

}

message HeartBeatReply {

}

service RaftRpc {
  rpc request_vote(VoteRequest) returns (VoteReply) {}
}
